modules/backendscheduler/provider/compaction.go

```diff
@@ -98,119 +98,47 @@ func (p *CompactionProvider) Start(ctx context.Context) <-chan *work.Job {
 		)
 
 		reset := func() {
-			job = nil
-			b.Reset()
+			p.curSelector = nil
```

Comment:
i find this code a bit confusing. for instance p.curTenantJob is reset on line 103, tested on 114 and incremented in createJob on 173 which makes it a bit tough to track how it's handled.

however, this PR is definitely moving things in the right direction. seeing some really solid simplification here. nice work. i think you're getting down to the core of it.

Reply:
In a couple PRs this might get a little cleaner also. :)